src/schema/users.ts

```diff
@@ -694,6 +706,48 @@ export const resolvers: IResolvers<any, Context> = {
       ctx: Context,
       info,
     ): Promise<Connection<GQLView>> => readHistoryResolver(args, ctx, info),
+    generateUniqueUsername: async (
+      _,
+      args: { name: string },
+      ctx: Context,
+    ): Promise<string> => {
+      const name = args.name.toLowerCase().replace(/[^a-zA-Z0-9]/g, '');
+
+      if (name.length === 0) {
+        return '';
+      }
+
+      const username = name.substring(0, 39);
+      let generatedUsernames = [username];
+
+      const generateRandomUsernames = (count: number) => {
```

Comment:
I don't think we need it as a function, we can just run this code. We don't reuse it anyway